uobtheatre/bookings/schema.py

```diff
@@ -281,7 +332,48 @@ def resolve_mutation(self, root, info, booking_id, price, nonce):
         return PayBooking(booking=booking, payment=payment)
 
 
+class CheckInBooking(AuthRequiredMixin, SafeMutation):
+    performance = graphene.Field("uobtheatre.productions.schema.PerformanceNode")
+    booking = graphene.Field(BookingNode)
+
+    class Arguments:
+        booking_reference = graphene.String(required=True)
+        performance_id = IdInputField(required=True)
+        tickets = graphene.List(CheckInTicketInput, required=True)
+
+    @classmethod
+    def resolve_mutation(self, root, info, booking_reference, tickets, performance_id):
+        performance = Performance.objects.get(id=performance_id)
+        booking = Booking.objects.get(reference=booking_reference)
+
+        # check if the booking pertains to the correct performance
+        if booking.performance == performance:
+            ticket_objects = list(map(lambda ticket: ticket.to_ticket(), tickets))
+            # loop through the ticket IDs given
+            for ticket in ticket_objects:
+                # Check the ticket booking matches the given booking
+                if ticket.booking == booking:
+                    ticket.check_in()
+                else:
+                    raise GQLFieldException(
+                        message="The ticket booking does not match the mutation booking."
+                    )
+                    # Raise ticket booking does not match the booking
+        else:
+            raise GQLFieldException(
+                message="The booking performance does not match the given performance."
+            )
```

Comment:
I would say its slightly easier to read when you don't use `else` and just lead with the exception. I.e.

```
        if booking.performance != performance:
            raise GQLFieldException
        do_other_things
```

Reply:
Right, because the exception will break out from the process flow! Will change.